### PR TITLE
refactor: IrcMessageHandler

### PR DESCRIPTION
**Pull request by pajlada**
# Description

The biggest change this does, is change previously-public API of IrcMessageHandler into internal/static APIs

The PR is best reviewed if you skip changes from https://github.com/Chatterino/chatterino2/pull/4927/commits/d6604de8dcc4d9b1a4c50ad32246ed7b6bec37f7 - potentially even reviewing commit-by-commit

### CODE REVIEW

File: src/providers/twitch/IrcMessageHandler.cpp [hunk 1]
```diff
@@ -507,129 +297,370 @@ void IrcMessageHandler::populateReply(
...
+        populateReply(tc, message, otherLoaded, builder);
+
+        if (!builder.isIgnored())
         {
-            server.mentionsChannel->addMessage(msg);
+            builtMessages.emplace_back(builder.build());
+            builder.triggerHighlights();
         }
+    }
+    else if (command == "USERNOTICE")
```
Comment (Nerixyz) [inline src/providers/twitch/IrcMessageHandler.cpp:598]: If there was an early return here, the `else if`s could be turned into `if`s.
```suggestion
        }
        return builtMessages;
    }

    if (command == "USERNOTICE")
```

File: src/providers/twitch/IrcMessageHandler.cpp [hunk 2]
```diff
@@ -907,98 +903,26 @@ void IrcMessageHandler::handleWhisperMessage(Communi::IrcMessage *message)
...
-            messageText = "Announcement";
-        }
-
-        auto b = MessageBuilder(systemMessage, parseTagString(messageText),
-                                calculateMessageTime(message).time());
-
-        b->flags.set(MessageFlag::Subscription);
-        auto newMessage = b.release();
-        builtMessages.emplace_back(newMessage);
+    if (_message->flags.has(MessageFlag::ShowInMentions))
```
Comment (Nerixyz) [inline src/providers/twitch/IrcMessageHandler.cpp:906]: `_message` seems a bit weird. Could this be renamed to something like `msg`, `built`, `builtMessage`, `whisper` or similar?

File: src/providers/twitch/IrcMessageHandler.cpp (hunk 1 above)
Comment (pajlada) [reply src/providers/twitch/IrcMessageHandler.cpp]: Resolved in [`c91f947` (#4927)](https://github.com/Chatterino/chatterino2/pull/4927/commits/c91f94744fc0d26c56ee23d4049b9d20eba9cd30), with bonus `u"xd"_s` fixes in [`441e50e` (#4927)](https://github.com/Chatterino/chatterino2/pull/4927/commits/441e50e300de8272109cec4a73584c4682cdc942)

File: src/providers/twitch/IrcMessageHandler.cpp (hunk 2 above)
Comment (pajlada) [reply src/providers/twitch/IrcMessageHandler.cpp]: Resolved in [`b772e3d` (#4927)](https://github.com/Chatterino/chatterino2/pull/4927/commits/b772e3dfd3ebaad21bd1111841297185d57bb479)

I opted to rename the parameter from `message` to `ircMessage`, so this function variable could be renamed to `message`

**Pull request merged**